.github/workflows/playwright.yml

```diff
@@ -153,8 +153,63 @@ jobs:
           path: playwright-report
           retention-days: 2
 
+  publish-report:
+    if: ${{ !cancelled() }}
+    needs: [merge-reports]
+    runs-on: ubuntu-latest
+    outputs:
+      page_url: ${{ steps.set-url.outputs.page_url }}
+    steps:
+      - uses: actions/checkout@v4
+      - name: Download HTML report
+        uses: actions/download-artifact@v4
+        with:
+          name: html-report--attempt-${{ github.run_attempt }}
+          path: playwright-report
+      - name: Determine report destination
+        id: report-dest
+        run: |
+          if [ "${{ github.event_name }}" = "pull_request" ]; then
+            DEST_DIR="${{ github.event.pull_request.number }}"
+          else
+            DEST_DIR="${{ github.run_id }}"
+          fi
+          echo "destination=$DEST_DIR" >> "$GITHUB_OUTPUT"
+          echo "Report will be published to: $DEST_DIR"
+      - name: Deploy report to GitHub Pages
+        if: always()
+        uses: peaceiris/actions-gh-pages@v4
```

Comment:
For actions that are not github's, it's recommended to pin those to the full length sha
ref: https://docs.github.com/en/actions/reference/security/secure-use#:~:text=Pin%20actions%20to%20a%20full%2Dlength%20commit%20SHA